tools/build/pack_windows.py

```diff
@@ -0,0 +1,838 @@
+    if not app_name:
+        app_name = os.path.basename(app_dir)
+
+    app_src = os.path.join(BUILD_ROOT_SRC_SUB_APP, app_name)
+
+    # copy source to  BUILD_ROOT_SRC_SUB_APP/pkg_name/app
+    if buildSRC(app_dir, os.path.join(app_src, "app"), build_json):
+        app_dest = safelyGetValue(build_json, "install-path")
+        if app_dest:
+            app_dest = os.path.join(app_dest_default, app_dest)
+        else:
+            app_dest = app_dest_default
+
+        if safelyGetValue(build_json, "all-apps") == "true":
+            app_dirs = os.listdir(app_src)
+            apps_num = 0
+            for i_app_dir in app_dirs:
+                if os.path.isdir(os.path.join(app_src, i_app_dir)):
+                    print "i_app_dir:" + i_app_dir
+                    i_app_name = os.path.basename(i_app_dir)
+                    print "i_app_name:" + i_app_name
+                    if not packAPP(
+                            build_json, os.path.join(app_src, i_app_name),
+                            app_dest, i_app_name):
+                        return False
+                    else:
+                        apps_num = apps_num + 1
+            if apps_num > 0:
+                LOG.info("Totally packed %d apps in %s" % (apps_num, app_dir))
+                return True
+        else:
+            return packAPP(build_json, app_src, app_dest, app_name)
+    return False
+
+
+def buildPKGAPP(build_json=None):
+    try:
+        LOG.info("+Building package APP ...")
+        if not os.path.exists(
+                os.path.join(BUILD_ROOT_SRC, "crosswalk-app-tools")):
+            if not doCopy(os.path.join(BUILD_PARAMETERS.pkgpacktools, "crosswalk-app-tools"),
+                          os.path.join(BUILD_ROOT_SRC, "crosswalk-app-tools")):
+                return False
+        if not os.path.exists(
+                os.path.join(BUILD_ROOT_SRC, "crosswalk.zip")):
+            if not doCopy(os.path.join(BUILD_PARAMETERS.pkgpacktools, "crosswalk.zip"),
+                          os.path.join(BUILD_ROOT_SRC, "crosswalk.zip")):
+                return False
+        os.chdir(BUILD_ROOT_SRC)
+
+        pkg_name = "org.test." + PKG_NAME.replace("-", "")
+        pack_cmd = "crosswalk-app-tools/src/crosswalk-app create " + pkg_name + " --platform=windows --windows-crosswalk=crosswalk.zip"
+        orig_dir = os.getcwd()
+        LOG.info(
+            "+Change dir to %s: " %
+            os.path.join(
+                BUILD_ROOT_SRC,
+                "crosswalk-app-tools"))
+        os.chdir(os.path.join(BUILD_ROOT_SRC, "crosswalk-app-tools"))
+        if not doCMD(pack_cmd, DEFAULT_CMD_TIMEOUT):
+            os.chdir(orig_dir)
+            return False
+
+        if not doCopy(os.path.join(BUILD_ROOT_SRC, "crosswalk-app-tools", pkg_name),
+                      os.path.join(BUILD_ROOT_SRC_PKG_APP, pkg_name)):
+            return False
+
+        if not doCopy(os.path.join(BUILD_ROOT_SRC, "icon.png"),
```

Comment:
Currently ".png" format icon is not supported in windows yet. Please use ".ico" format instead.